ui/src/main/java/com/adyen/checkout/ui/internal/Images.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 15/8/2025.
+ */
+
+package com.adyen.checkout.ui.internal
+
+import android.graphics.Bitmap
+import androidx.annotation.DrawableRes
+import androidx.annotation.RestrictTo
+import androidx.compose.foundation.Image
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import com.adyen.checkout.test.R
+
+/**
+ * Represents the state of an image loading operation.
+ */
+private sealed class ImageLoadState {
+    object Loading : ImageLoadState()
+    data class Success(val bitmap: Bitmap) : ImageLoadState()
+    object Error : ImageLoadState()
+}
+
+/**
+ * Composable that loads an image from a URL using the provided ImageLoader.
+ *
+ * This is the core composable that handles the state management of the image loading process.
+ *
+ * @param url The URL of the image to load.
+ * @param modifier The modifier to be applied to the Image.
+ * @param contentDescription The content description for accessibility.
+ * @param imageLoader The ImageLoader instance to use. Defaults to the one provided by the Context.
+ * @param placeholder A painter to show while the image is loading.
+ * @param errorFallback A painter to show if the image fails to load.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+@Composable
+fun NetworkImage(
+    url: String,
+    modifier: Modifier = Modifier,
+    contentDescription: String?,
+    imageLoader: ImageLoader,
```

Comment:
Nit: since `imageLoader` has no default value it should be placed above this `modifier`